src/main/java/org/folio/circulation/services/FeeFineService.java

```diff
@@ -1,58 +1,167 @@
 package org.folio.circulation.services;
 
 import static java.util.concurrent.CompletableFuture.allOf;
+import static java.util.concurrent.CompletableFuture.completedFuture;
+import static org.folio.circulation.services.feefine.FeeRefundProcessor.createLostItemFeeRefundProcessor;
+import static org.folio.circulation.support.Result.combined;
+import static org.folio.circulation.support.Result.failed;
+import static org.folio.circulation.support.Result.succeeded;
 
 import java.util.Collection;
+import java.util.List;
 import java.util.concurrent.CompletableFuture;
 
+import org.apache.commons.lang3.tuple.ImmutablePair;
+import org.apache.commons.lang3.tuple.Pair;
+import org.folio.circulation.domain.Account;
 import org.folio.circulation.domain.AccountRepository;
-import org.folio.circulation.domain.FeeFineAccountAndAction;
 import org.folio.circulation.domain.FeeFineActionRepository;
-import org.folio.circulation.domain.representations.AccountStorageRepresentation;
-import org.folio.circulation.domain.representations.FeeFineActionStorageRepresentation;
+import org.folio.circulation.domain.ServicePoint;
+import org.folio.circulation.domain.ServicePointRepository;
+import org.folio.circulation.domain.User;
+import org.folio.circulation.domain.UserRepository;
+import org.folio.circulation.domain.representations.StoredAccount;
+import org.folio.circulation.domain.representations.StoredFeeFineAction;
+import org.folio.circulation.services.feefine.AccountRefundContext;
+import org.folio.circulation.services.feefine.FeeRefundProcessor;
+import org.folio.circulation.services.support.CreateAccountCommand;
+import org.folio.circulation.services.support.RefundAccountCommand;
 import org.folio.circulation.support.Clients;
 import org.folio.circulation.support.Result;
+import org.folio.circulation.support.ServerErrorFailure;
 import org.folio.circulation.support.results.CommonFailures;
 
+import io.vertx.core.json.JsonObject;
+
 public class FeeFineService {
   private final AccountRepository accountRepository;
   private final FeeFineActionRepository feeFineActionRepository;
+  private final UserRepository userRepository;
+  private final ServicePointRepository servicePointRepository;
+  private final FeeRefundProcessor lostItemRefundProcessor;
 
   public FeeFineService(Clients clients) {
     this.accountRepository = new AccountRepository(clients);
     this.feeFineActionRepository = new FeeFineActionRepository(clients);
+    this.userRepository = new UserRepository(clients);
+    this.servicePointRepository = new ServicePointRepository(clients);
+    this.lostItemRefundProcessor = createLostItemFeeRefundProcessor();
   }
 
   public CompletableFuture<Result<Void>> createAccountsAndActions(
```

Comment:
Does the client to this method need to know that actions are also being created?

What is the client's intent? Is it to charge one or more fees? Does this also charge fines?

```suggestion
  public CompletableFuture<Result<Void>> createAccounts(
```

Reply:
the `AndActions` part was added just to let know that actions is handled automatically by the code.

I thnk it can be renamed to `createAccounts`.

> Does this also charge fines?

I'm not sure if there is any difference between how account for fee vs fine created.

Reply:
> I'm not sure if there is any difference between how account for fee vs fine created.

The intent seems to be to move away from the account terminology (see comments in mod-feefines pull request about Vega reviewing the domain model). 

I'm trying to understand if this same method is used for fees and fines, as that might influence the name if we decide to remove the term account from it.

Reply:
Are you suggesting to rename the method to something `createFeesFines` instead of using `account` term?

Reply:
> Are you suggesting to rename the method to something `createFeesFines` instead of using `account` term?

I think there is a move away from the account term in general. I'm not keen on it in circulation because it isn't a part of the domain language of circulation. From the conversations and stories I've read that is centred on charging a fee or a fine.

I'm trying to understand how the method is used in order to decide if there might be a more appropriate name.

If it is intended to create fees or fines then that may present a bit of a challenge, as I'm not keen on names like `createFeeOrFine` as it tends to reduce clarity rather than improve it. I'm not aware of a collective term for fee and fine together.

Reply:
I'd postpone rename then, since it is not 100% clear about new domain terminology. Is it fine?

Reply:
> I'd postpone rename then, since it is not 100% clear about new domain terminology. Is it fine?

Sure, let's leave this for now